ppr-api/src/ppr_api/models/client_code.py

```diff
@@ -95,6 +95,41 @@ def find_by_head_office_code(cls, head_office_id: str = None):
 
         return party_codes
 
+    @classmethod
+    def find_by_account_number(cls, account_number: int):
+        """Return a list of client parties searching by account number."""
+        party_codes = []
+        if account_number:
+            party_list = db.session.query(ClientCode).filter(ClientCode.bconline_account == account_number).all()
+            if not party_list:
+                return party_codes
+            for party in party_list:
+                party_codes.append(party.json)
+
+        return party_codes
+
+    @classmethod
+    def find_by_branch_start(cls, branch_code: str):
+        """Return a list of client parties matching on branch ids that start with a query number."""
+        party_codes = []
+        if not branch_code or not branch_code.strip().isdigit():
+            return party_codes
+        query_num = branch_code.strip()
+        if int(query_num) < 100:  # Exact match
+            party = cls.find_by_code(query_num)
+            if party:
+                party_codes.append(party)
```

Comment:
do you need to append party.json here?

Reply:
Yes, this condition is an exact match returning either a single entry or an empty array.